sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -1894,6 +1894,149 @@ static Task WritePartitionAsync(Response<BlobDownloadInfo> response, Stream dest
+            DeleteSnapshotsOption? deleteOptions = default,
+            BlobAccessConditions? accessConditions = default,
+            CancellationToken cancellationToken = default) =>
+            await DeleteIfExistsInternal(
+                deleteOptions,
+                accessConditions ?? default,
+                true, // async
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// The <see cref="DeleteIfExistsInternal"/> operation marks the specified blob
+        /// or snapshot for deletion, if the blob exists. The blob is later deleted
+        /// during garbage collection.
+        ///
+        /// Note that in order to delete a blob, you must delete all of its
+        /// snapshots. You can delete both at the same time using
+        /// <see cref="DeleteSnapshotsOption.Include"/>.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/rest/api/storageservices/delete-blob" />.
+        /// </summary>
+        /// <param name="deleteOptions">
+        /// Specifies options for deleting blob snapshots.
+        /// </param>
+        /// <param name="accessConditions">
+        /// Optional <see cref="BlobAccessConditions"/> to add conditions on
+        /// deleting this blob.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> on successfully deleting.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<bool>> DeleteIfExistsInternal(
+            DeleteSnapshotsOption? deleteOptions,
+            BlobAccessConditions accessConditions,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            try
+            {
+                Response response = await DeleteInternal(
+                    deleteOptions,
+                    accessConditions,
+                    async,
+                    cancellationToken,
+                    Constants.Blob.Base.DeleteIfExists)
+                    .ConfigureAwait(false);
+                return Response.FromValue(response, true);
+            }
+            catch (StorageRequestFailedException storageRequestFailedException)
+            {
+                //If the blob already exists, a StorageRequestFailure with be thrown
+                if (storageRequestFailedException.ErrorCode == Constants.Blob.NotFound)
+                {
+                    return Response.FromValue(default, false);
```

Comment:
It's fine to check this in, but I think it's going to be difficult for people to use.  A `default` `Response` is going to be pretty confusing.  I'd rather provide people with the actual response that failed and we may need to do a little work to thread it around so you can get it from the exception.  I'll look at that as part of  #7896 as well.